applications/StructuralMechanicsApplication/custom_utilities/constitutive_law_utilities.cpp

```diff
@@ -173,17 +173,110 @@ void ConstitutiveLawUtilities<TVoigtSize>::CalculateLodeAngle(
 /***********************************************************************************/
 /***********************************************************************************/
 
+template<SizeType TVoigtSize>
+void ConstitutiveLawUtilities<TVoigtSize>::CalculateGreenLagrangianStrain(
+    const MatrixType& rCauchyTensor,
+    Vector& rStrainVector
+    )
+{
+    rStrainVector[0] = 0.5 * ( rCauchyTensor( 0, 0 ) - 1.00 ); // xx
+    rStrainVector[1] = 0.5 * ( rCauchyTensor( 1, 1 ) - 1.00 ); // yy
+    rStrainVector[2] = 0.5 * ( rCauchyTensor( 2, 2 ) - 1.00 ); // zz
+    rStrainVector[3] = rCauchyTensor( 0, 1 ); // xy
+    rStrainVector[4] = rCauchyTensor( 1, 2 ); // yz
+    rStrainVector[5] = rCauchyTensor( 0, 2 ); // xz
+}
```

Comment:
This is only valid for TVoigSize = 6.
Since the template allows different Voigt sizes, it would be worth to consider them.

Reply:
Yes it was pending to implement, I can fix it

Reply:
Good!